cassandra/io/twistedreactor.py

```diff
@@ -133,6 +135,25 @@ def _cleanup(self):
                             "Cluster.shutdown() to avoid this.")
             log.debug("Event loop thread was joined")
 
+    def add_timer(self, timer):
+        self._timers.add_timer(timer)
+        reactor.callFromThread(self._schedule_timeout, timer.end)
```

Comment:
To my knowledge this is the only way to interact with the loop thread in a safe way (similar to async events in libev). I callFromThread to get into the scheduling function with the loop thread, where it is safe to modify the loop timer.
If there are better techniques I'm open to suggestions.